src/frontend/sharedComponents/GPSPill.tsx

```diff
@@ -1,98 +1,85 @@
-import React, {FC, useMemo} from 'react';
-import {StyleSheet, TouchableOpacity, View} from 'react-native';
-import {Text} from './Text';
-import {ParamListBase, useIsFocused} from '@react-navigation/native';
+import React, {FC} from 'react';
+import {StyleSheet, TouchableOpacity} from 'react-native';
+import {DARK_GREY, WHITE} from '../lib/styles';
+import {GpsErrorIcon, GpsSearchingIcon, GpsGoodIcon} from './icons';
+import {useSharedLocationContext} from '../contexts/SharedLocationContext';
 import {useLocationProviderStatus} from '../hooks/useLocationProviderStatus';
 import {getLocationStatus} from '../lib/utils';
-import {defineMessages, useIntl} from 'react-intl';
-import {GpsIcon} from './icons';
-import {useSharedLocationContext} from '../contexts/SharedLocationContext';
-import {BLACK, WHITE} from '../lib/styles';
-import {BottomTabNavigationProp} from '@react-navigation/bottom-tabs';
+import {BodyText} from './Text/BodyText';
+import {UIActivityIndicator} from 'react-native-indicators';
 
-const m = defineMessages({
-  noGps: {
-    id: 'sharedComponents.GpsPill.noGps',
-    defaultMessage: 'No GPS',
-  },
-  searching: {
-    id: 'sharedComponents.GpsPill.searching',
-    defaultMessage: 'Searching…',
-  },
-});
-
-interface GPSPill {
-  navigation: BottomTabNavigationProp<ParamListBase, string, undefined>;
-}
+type GPSPillProps = {
+  onPress?: () => void;
+};
 
-export const GPSPill: FC<GPSPill> = ({navigation}) => {
-  const isFocused = useIsFocused();
-  const {formatMessage: t} = useIntl();
-  const {locationState, fgPermissions} = useSharedLocationContext();
+export const GPSPill: FC<GPSPillProps> = ({onPress}) => {
   const locationProviderStatus = useLocationProviderStatus();
```

Comment:
Can you pass this locationProviderStatus to GPS pill and get rid of useLocationProviderStatus in the gps pill. They are both starting 2 separate subscriptions of the same thing